src/fosslight_util/download.py

```
@@ -822,7 +1090,137 @@ def download_git_clone(git_url, target_dir, checkout_to="", tag="", branch="",
+        raise
+    except Exception:
+        pass
+
+    before = set()
+    try:
+        before = {p.resolve() for p in Path(target_dir).iterdir()}
+    except OSError:
+        before = set()
+
+    try:
+        proc = subprocess.Popen(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)